src/daemon/pex/pex.go

```diff
@@ -360,7 +360,23 @@ func (px *Pex) AddPeer(addr string) error {
 	}
 
 	if px.Config.Max > 0 && px.peerlist.len() >= px.Config.Max {
-		return ErrPeerlistFull
+
+		i := float64(0)
```

Comment:
This block of code should be rewritten to find the first peer that exceeds the time threshold and remove them.

The threshold should be either a configured property of `Pex` or a global const, and measured in seconds